.github/workflows/python-ci-docs.yml

```diff
@@ -40,8 +40,6 @@ jobs:
         with:
           python-version: 3.12
       - name: Install
-        working-directory: ./mkdocs
-        run: pip install -r requirements.txt
-      - name: Build
-        working-directory: ./mkdocs
-        run: mkdocs build --strict
+        run: make install
```

Comment:
This will probably slow down the runs of the docs builds (since we install everything), do you have any numbers?

Reply:
action run for this PR took only `1m 34s`
https://github.com/apache/iceberg-python/actions/runs/12623916115?pr=1486

on main this action usually takes ~12m 
https://github.com/apache/iceberg-python/actions/workflows/python-ci.yml?query=branch%3Amain

Reply:
im not sure if this is apple-to-apple.
alternatively, we can add the mkdoc deps as a new poetry group and just install those
```
poetry install --docs 
```